test_files/clutz.no_externs/import_default.js

```diff
@@ -0,0 +1,18 @@
+/**
+ *
+ * @fileoverview Reproduces a problem where a renamed Clutz default export ({default as X}) would
+ * produce type annotations including an indirection to the aliased symbol.
+ *
+ * Generated from: test_files/clutz.no_externs/import_default.ts
+ * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
+ */
+goog.module('test_files.clutz.no_externs.import_default');
+var module = module || { id: 'test_files/clutz.no_externs/import_default.ts' };
+module = module;
```

Comment:
Time for me to learn something, I think.

Why is this `module = module;` assignment here?

Reply:
Good question, and it's a bit involved.

Some code in Angular requires a module bundler that exposes a `module.id` to the local file, containing the current module's path (not ID!). I.e. it depends on this API: https://nodejs.org/api/modules.html#modules_module_id. That's why there's a `var module = {id: ...}`.

Regarding the `var module = module || ...` bit, I no longer have full context on why that was needed. The code (in `googmodule.ts`) only comments `The following pattern ensures closure doesn't throw an error in advanced optimizations mode.`.

As for the last bit, `module = module`, this is to prevent an unused variable warning if `module` turns out not to be used. This is to accommodate projects that turn lint warnings into errors (which of course is a bad idea, but what can you do...). I think alternatively we could use `unusedLocalVariables` for the entire file, which would probably also be OK. We might not have been aware of that when we put this in, or might not have had the tech to reliably insert of fileoverview comment at the time.